astropy/coordinates/angles/core.py

```diff
@@ -377,23 +373,16 @@ def to_string(
         def do_format(val):
             # Check if value is not nan to avoid ValueErrors when turning it into
             # a hexagesimal string.
-            if not np.isnan(val):
-                s = func(float(val))
-                if alwayssign and not s.startswith("-"):
-                    s = "+" + s
-                if format == "latex" or format == "latex_inline":
-                    s = f"${s}$"
-                return s
-            s = f"{val}"
-            return s
-
-        values = self.to_value(unit)
-        format_ufunc = np.vectorize(do_format, otypes=["U"])
-        result = format_ufunc(values)
+            if np.isnan(val):
+                return "nan"
```

Comment:
This is not quite right if the `Angle` is complex - though really the rest of the code breaks for that case too... so, for this refactoring PR it doesn't matter. 

Aside: It is also slightly bizarre that no `$` are used for latex, but that's even more out of scope here. Really, we need to use `Quantity.to_string()`...